SeqVision/modules/dna_rna_tools_module.py

```diff
@@ -0,0 +1,95 @@
+set_rna = {"A", "G", "C", "U", "a", "g", "c", "u"}
+set_dna = {"A", "G", "C", "T", "a", "g", "c", "t"}
+ComplementDictDna = {
+    "a": "t",
+    "A": "T",
+    "t": "a",
+    "T": "A",
+    "g": "c",
+    "G": "C",
+    "c": "g",
+    "C": "G",
+}
+ComplementDictRna = {
+    "a": "u",
+    "A": "U",
+    "u": "a",
+    "U": "A",
+    "g": "c",
+    "G": "C",
+    "c": "g",
+    "C": "G",
+}
+TranscribeDictDna = {
+    "a": "a",
+    "A": "A",
+    "t": "u",
+    "T": "U",
+    "g": "g",
+    "G": "G",
+    "c": "c",
+    "C": "C",
+}
+TranscribeDictRna = {
+    "a": "a",
+    "A": "A",
+    "u": "t",
+    "U": "T",
+    "g": "g",
+    "G": "G",
+    "c": "c",
+    "C": "C",
+}
+
+
+def is_rna(seq: str):
+    return set(seq.upper()) <= set_rna
+
+
+def is_dna(seq: str):
+    return set(seq.upper()) <= set_dna
+
+
+def complement(seq: str):
+    complement_seq = ""
+    if is_dna(seq):
+        complement_seq = "".join([ComplementDictDna[i] for i in seq])
+    if is_rna(seq):
+        for i in seq:
+            complement_seq = "".join([ComplementDictRna[i] for i in seq])
+    return complement_seq
```

Comment:
А если пользователь передаст "гибрид-последовательность", например, AUGT. Я бы добавила проверку на такие несуществующие последовательности

```suggestion
    def get_sequence_type(seq: str):
    """Determine type of molecule."""
    is_dna_flag = is_dna(seq)
    is_rna_flag = is_rna(seq)
    if is_dna_flag and not is_rna_flag:
        return "dna"
    elif is_rna_flag and not is_dna_flag:
        return "rna"
    elif is_dna_flag and is_rna_flag:
        return "mixed"
    else:
        return "invalid"


def complement(seq: str):
    """Returns complement sequence."""
    seq_type = determine_sequence_type(seq)
    if seq_type == "dna":
        return "".join([ComplementDictDna[i] for i in seq])
    elif seq_type == "rna":
        return "".join([ComplementDictRna[i] for i in seq])
    elif seq_type == "mixed":
        raise ValueError("The sequence contains both RNA and DNA bases.")
    else:
        raise ValueError("Invalid sequence: contains unknown characters.")
```